src/components/App/App.js

```diff
@@ -24,37 +25,74 @@ class App extends Component {
   }
 
   addPokemon = (queriedPokemon) => {
-    this.validatePokemonData(queriedPokemon)
+    const foundPokemon = this.validatePokemonData(queriedPokemon)
+    this.setState({
+      foundPokemon: [foundPokemon]
+    })
     //set state with the validated pokemon
   }
 
   validatePokemonData = (queriedPokemon) => {
     const lowerCaseInput = queriedPokemon.queriedPokemon.toLowerCase()
-
-    const verifiedName = this.state.pokeDex.find(pokemon => {
+    
+    const verifiedName = this.state.pokeDex.find((pokemon, index) => {
       let lowerCaseName = pokemon.name.toLowerCase()
-      if (lowerCaseName.includes(lowerCaseInput) && lowerCaseInput !== '')  {
+
+      if (lowerCaseName.includes(lowerCaseInput) && lowerCaseInput !== '' && lowerCaseInput !== undefined) {
         return pokemon
+      } else if (parseInt(lowerCaseInput)){
+        if (index+1 === parseInt(lowerCaseInput)){
+          return pokemon
+        }
       }
-    })
+    });
+
+    console.log(verifiedName)
 
     if (verifiedName === undefined) {
-      console.log('No Good NAMe!')
-    } else {
+      return this.setState({ error: "Not  a valid name, try again" })
+      console.log('No Good Name!')
+    } else { 
       console.log(verifiedName, 'IT WORKSSSSSSS!!!!')
       return verifiedName
     }
+   
   }
 
   // pass the validated query through addPokemon
 
+  getPokemonImage = (id) => {
+    
+    let pokemonImage = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${id}.png`
+   
+    return pokemonImage
+
+  }
+
   render() {
+
+    const text = 'Loading...';
+
     return(
-      <div>Hello World
+      <div> Welcome to PoKeDeX, are you ready to catch your Pokemon?
         <Search addPokemon={this.addPokemon}/>
+        {(this.state.foundPokemon.length === 0 && !this.state.error) && <h2>{ text }</h2>}
+        {(this.state.error && <h2> { this.state.error }</h2>)}
+        {(this.state.foundPokemon.length !== 0 && !this.state.error)&& 
+        <PokedexGrid pokedexData={this.state.foundPokemon} getPokemonImage={this.getPokemonImage}/>}
+        {(this.state.foundPokemon.length === 0) && 
+        <PokedexGrid pokedexData={this.state.pokeDex} getPokemonImage={this.getPokemonImage}/>
+        }
```

Comment:
Great conditional rendering!

Reply:
Beautiful. Great uses of `&&`.

Reply:
> Great conditional rendering!

Hi Matt here is where I'm using the `foundPokemon.length `to be able to render what it needs to be rendered in the page